### PR TITLE
Remove old _knn_search tech preview API in v9

### PR DESCRIPTION
**Pull request by benwtrent**
Removes the old `_knn_search` API that was never out of tech preview and deprecated throughout the v8 cycle. 

To utilize the API, `compatible-with=8` can be utilized.

### CODE REVIEW

**Review by john-wagster — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@elastic/docs This removes a documentation page as the API is no longer generally accessible in v9. 

Do we need to auto-forward the page? Can we do that in our docs?

File: docs/reference/redirects.asciidoc [hunk 1]
```diff
@@ -1942,3 +1942,8 @@ Refer to <<get-ip-location-database-api>>.
 === Delete geoip database configuration API
 
 Refer to <<delete-ip-location-database-api>>.
+
+[role="exolude",id="knn-search-api"]
```
Comment (shainaraskas) [inline docs/reference/redirects.asciidoc:1946]:
```suggestion
[role="exclude",id="knn-search-api"]
```

Comment (benwtrent) [reply docs/reference/redirects.asciidoc:1946]: LOL thanks!

File: server/src/main/java/org/elasticsearch/rest/action/search/RestKnnSearchAction.java [hunk 2]
```diff
@@ -34,19 +34,11 @@ public class RestKnnSearchAction extends BaseRestHandler {
 
     public RestKnnSearchAction() {}
 
-    @UpdateForV9(owner = UpdateForV9.Owner.SEARCH_RELEVANCE)
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/rest/action/search/RestKnnSearchAction.java:37]: Should we add a comment to remove this handler for V10?

**Review by mayya-sharipova — APPROVED** [review]
Thanks Ben, nice code cleanup

**Comment by leemthompo** [comment]
@benwtrent is this PR relevant to the serverless changelog? [FYI this question is based on [9.0 breaking changes](https://staging-website.elastic.co/docs/release-notes/elasticsearch/breaking-changes)]

**Pull request merged**